File: 566.reshape-the-matrix.py
#
# @lc app=leetcode id=566 lang=python
#
# [566] Reshape the Matrix
#

# @lc code=start
class Solution(object):
    def matrixReshape(self, mat, r, c):
        """
        :type mat: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """

        if r * c != len(mat) * len(mat[0]):
            return mat

        return [mat[i][j] for i in range(len(mat)) for j in range(len(mat[0]))].reshape(r, c)

# @lc code=end

File: 566.reshape-the-matrix.js
/*
 * @lc app=leetcode id=566 lang=javascript
 *
 * [566] Reshape the Matrix
 */

// @lc code=start
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
    const result = []
    const arr = mat.flat()
    if (r * c !== arr.length) return mat
    for (let i = 0; i < r; i++) {
        result.push(arr.splice(0, c))
    }
    return result
};
// @lc code=end
